concent_api/verifier/tasks.py

```diff
@@ -144,27 +152,125 @@ def blender_verification_order(
                 subtask_id,
                 VerificationResult.ERROR,
                 str(completed_process.stderr),
-                'verifier.blender_verification_order.running_blender'
+                ErrorCode.VERIFIIER_RUNNING_BLENDER_FAILED.name
             )
             return
-    except SubprocessError as e:
+    except SubprocessError as exception:
         verification_result.delay(
             subtask_id,
             VerificationResult.ERROR,
-            str(e),
-            'verifier.blender_verification_order.running_blender'
+            str(exception),
+            ErrorCode.VERIFIIER_RUNNING_BLENDER_FAILED.name
         )
         return
 
     # Verifier deletes source files of the Blender project from its storage.
     # At this point there must be source files in VERIFIER_STORAGE_PATH otherwise verification should fail before.
     source_files_list = get_files_list_from_archive(
-        os.path.join(settings.VERIFIER_STORAGE_PATH, source_package_path)
+        generate_verifier_storage_file_path(source_package_path)
     )
     for file_path in source_files_list + [source_package_path]:
         delete_file(file_path)
 
-    verification_result.delay(
-        subtask_id,
-        VerificationResult.MATCH.name,
-    )
+    blender_output_file_name = generate_blender_output_file_name(scene_file)
+    upload_file_name = generate_upload_file_name(subtask_id, output_format)
+
+    upload_file_content = None
+
+    try:
+        with open(generate_verifier_storage_file_path(blender_output_file_name), 'r') as upload_file:
+            upload_file_content = upload_file.read()
+    except OSError as exception:
+        logger.error(f'Uploading blender generated file to storage failed with: {exception}')
+        sentry_logger.error(str(exception))
+    except MemoryError as exception:
+        logger.error(f'Loading result files into memory failed with: {exception}')
+        verification_result.delay(
+            subtask_id,
+            VerificationResult.ERROR,
+            str(exception),
+            ErrorCode.VERIFIIER_LOADING_FILES_INTO_MEMORY_FAILED.name
+        )
+        return
+
+    if upload_file_content is not None:
```

Comment:
I would prefer not to have `None` based checks... This whole `if` block should be moved below line `182`

Reply:
@dybi I have followed @cameel comment in https://github.com/golemfactory/concent/pull/318#discussion_r195136617

```
This should be outside of the try ... except block. An exception handler should cover only the code where you expect the exception.
```

Personally I think both of you are right in this case, but using `None` is less evil. I have moved place where variable is created to `expect` block